nixos/modules/config/malloc.nix

```diff
@@ -22,6 +22,15 @@ let
       '';
     };
 
+    jemalloc-mariadb = {
+      libPath = "${pkgs.jemalloc-mariadb}/lib/libjemalloc.so";
+      description = ''
+        A general purpose allocator that emphasizes fragmentation avoidance
+        and scalable concurrency support. This version of jemalloc is
+        compatible with mariadb.
```

Comment:
```suggestion
        A general purpose allocator that emphasizes fragmentation avoidance
        and scalable concurrency support.
        MariaDB versions before 10.5.1 need a special version of jemalloc, otherwise they'll crash on start.
```